lib/stats/stats-cluster.c

```diff
@@ -22,10 +22,85 @@
  *
  */
 #include "stats/stats-cluster.h"
+#include "mainloop.h"
 
 #include <assert.h>
 #include <string.h>
 
+GPtrArray *stats_types;
+
+int
+scs_static_const(guint constant, guint action, guint data)
+{
+  static guint constant_storage[SCS_STATIC_MAX];
+  g_assert(constant < SCS_STATIC_MAX);
+
+  switch (action)
+    {
+    case SCS_ACTION_GET:
+      g_assert(constant_storage[constant]); // Assert for use before init
+      return constant_storage[constant];
+    case SCS_ACTION_SET:
+      g_assert(data);
+      g_assert(data < SCS_SOURCE_MASK);
+      constant_storage[constant] = data;
+      return 0;
+    default:
+      g_assert_not_reached();
+    }
+
+  return 0;
+}
+
```

Comment:
I don't see why we don't split this function into a separate set and get function? the only reason I can think of is that it contains the constant_storage as a static variable. Pushing that into the global scope (but keeping it static) would get rid off SCS_ACTION_GET/SET and make this pretty complex function two simple ones. I think that's a fair price to pay for the increase of visibility for the constant_storage variable.

Reply:
You are right. For now I reverted that patch.